src/screens/volunteer/VolunteerStreamScreen.tsx

```diff
@@ -0,0 +1,224 @@
+    openWebScreen,
+    queryVariables: { authToken, setIsCollapsed, setPostForModal, userGuid },
+    refetch
+  });
+
+  const renderItem = useCallback(
+    (props) => {
+      const objectModel = props?.item?.content?.metadata?.object_model;
+      switch (objectModel) {
+        case VolunteerObjectModelType.CALENDAR:
+          return renderCalendarItem(props);
+        case VolunteerObjectModelType.POST:
+          return renderPostItem(props);
+        default:
+          return null;
+      }
+    },
+    [renderCalendarItem, renderPostItem]
+  );
+
+  const onEndReached = useCallback(async () => {
+    if (hasNextPage) {
+      return await fetchNextPage();
+    }
+
+    return {};
+  }, [fetchNextPage, hasNextPage]);
+
+  const results = data?.pages?.flatMap((page) => page?.results)?.filter(Boolean) || [];
+
+  const volunteerData = useMemo(() => {
+    // Filter out deleted entries (state === 100)
+    // see states here: https://github.com/humhub/humhub/blob/master/protected/humhub/modules/content/models/Content.php#L133-L136
+    const activeEntries = results?.filter(
+      (item: { content?: { metadata?: { state?: number } } }) =>
+        item.content?.metadata?.state !== 100
+    );
+
+    // return keys and values are taken from `parseVolunteerData` in volunteerHelper
+    return activeEntries?.map((volunteer) => {
+      const objectModel = volunteer?.content?.metadata?.object_model;
+      let teaserTitle;
+
+      if (objectModel === VolunteerObjectModelType.CALENDAR) {
+        teaserTitle = volunteer.content?.topics?.map((topic) => topic.name).join(', ');
+      }
+
+      return {
+        ...volunteer,
+        title: volunteer.title || volunteer.name,
+        subtitle:
+          volunteer.subtitle ||
+          volunteerSubtitle(
+            volunteer,
+            objectModel === VolunteerObjectModelType.CALENDAR
+              ? QUERY_TYPES.VOLUNTEER.CALENDAR
+              : query,
+            true,
+            false
+          ),
+        badge: volunteer.badge,
+        statustitle: volunteer.statustitle,
+        statustitleIcon: volunteer.statustitleIcon,
+        teaserTitle,
+        picture: volunteer.picture,
+        routeName: ScreenName.VolunteerDetail,
+        onPress: volunteer.onPress,
+        listDate: volunteer.listDate || volunteerListDate(volunteer),
+        status: volunteer.status,
+        params: {
+          title: getTitleForQuery(
+            objectModel === VolunteerObjectModelType.CALENDAR ? QUERY_TYPES.VOLUNTEER.CALENDAR : '',
+            volunteer
+          ),
+          query:
+            objectModel === VolunteerObjectModelType.CALENDAR
+              ? QUERY_TYPES.VOLUNTEER.CALENDAR
+              : undefined,
+          queryVariables: { id: volunteer.user?.id ? `${volunteer.user.id}` : `${volunteer.id}` },
+          rootRouteName: ROOT_ROUTE_NAMES.VOLUNTEER,
+          details: volunteer
+        },
+        bottomDivider: false
+      };
+    });
+  }, [results]);
```

Comment:
Is the object here very long and complex? For example, if we already have a `badge` value within the `volunteer`, why is it added again as `badge: volunteer.badge`?